src/py21cmfast/_utils.py

```diff
@@ -1135,7 +1135,7 @@ def read(
         self,
         direc: Union[str, Path, None] = None,
         fname: Union[str, Path, None, h5py.File, h5py.Group] = None,
-        keys: Optional[Sequence[str]] = None,
+        keys: Optional[Sequence[str]] = [],
```

Comment:
Ah, you should not set the default as a mutable object -- make it instead an empty tuple.